feature/container/containerz/tests/container_lifecycle/containerz_test.go

```diff
@@ -65,5 +93,155 @@ func TestDeployAndStartContainer(t *testing.T) {
 	if _, err = cntrCli.Ping(ctx, &cpb.PingRequest{}); err != nil {
 		t.Errorf("unable to reach cntrsrv: %v", err)
 	}
+}
+
+// TestRetrieveLogs implements CNTR-1.2 validating that logs can be retrieved from a
+// running container.
+func TestRetrieveLogs(t *testing.T) {
+	ctx := context.Background()
+	cli := startContainer(ctx, t)
+	defer stopContainer(ctx, t, cli)
+
+	logCh, err := cli.Logs(ctx, instanceName, false)
+	if err != nil {
+		t.Errorf("unable to obtain logs for %s: %v", instanceName, err)
+	}
+
+	var logs []string
+	for msg := range logCh {
+		logs = append(logs, msg.Msg)
+		if msg.Error != nil {
+			t.Errorf("logs returned an error: %v", err)
+		}
+	}
+
+	if len(logs) == 0 {
+		t.Errorf("no logs were returned")
+	}
+}
+
+// TestListContainers implements CNTR-1.3 validating listing running containers.
+func TestListContainers(t *testing.T) {
+	ctx := context.Background()
+	cli := startContainer(ctx, t)
+	defer stopContainer(ctx, t, cli)
+
+	listCh, err := cli.ListContainer(ctx, true, 0, nil)
+	if err != nil {
+		t.Errorf("unable to list containers: %v", err)
+	}
+
+	wantCntrs := []string{"cntrsrv:latest"}
+	var gotCntrs []string
+
+	for cnt := range listCh {
+		gotCntrs = append(gotCntrs, cnt.ImageName)
+	}
+
+	if diff := cmp.Diff(wantCntrs, gotCntrs, cmpopts.SortSlices(func(l, r string) bool { return l < r })); diff != "" {
+		t.Errorf("ListContainer() returned diff (-want, +got):\n%s", diff)
+	}
+}
+
+// TestStopContainer implements CNTR-1.4 validating that stopping a container works as expected.
+func TestStopContainer(t *testing.T) {
+	ctx := context.Background()
+	cli := startContainer(ctx, t)
+	stopContainer(ctx, t, cli)
+
+	// wait for container to stop
+	time.Sleep(2 * time.Second)
+
+	listCh, err := cli.ListContainer(ctx, true, 0, nil)
+	if err != nil {
+		t.Errorf("unable to list containers: %v", err)
+	}
 
+	for cntr := range listCh {
+		t.Errorf("StopContainer did not stop the container: %v", cntr)
+	}
+}
+
+func TestVolumes(t *testing.T) {
```

Comment:
Can you add a comment here? I think it'd be worth just noting that the test here ensures that a volume can be created and removed from an API perspective -- it doesn't check that it's possible to actually _use_ that volume.